t/python/t0007-watchers.py

```diff
@@ -206,6 +207,33 @@ def timer_cb(handle, watcher, revents, _args):
                 self.assertTrue(rc < 0)
 
 
+class TestIssue3973(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.f = flux.Flux()
+
+    def test_watcher_gc(self):
+
+        cb_called = [0]
```

Comment:
This is the right thing to do for Python 2, and I know it is in some of the other tests in this module, but in Python 3 the standard thing to do is to use the ``nonlocal`` keyword instead of wrapping the value in a list. Something like:

```python
cb_called = 0

    def cb(handle, watcher, revents, _args):
          nonlocal cb_called
          cb_called += 1
```

Reply:
Thanks! I was completely unaware and just copied other examples in the test (which were probably from the time of Python2)